File: code_samples/javascript/antigravity_api.js
/**
 * Antigravity Automation API — JavaScript (Node.js) Code Samples
 * Works with Node 18+ (native fetch) or install: npm install node-fetch ws
 */

const BASE_URL = "http://localhost:5000";

// ── Automation ──────────────────────────────────────

async function toggleAutoRun() {
    const res = await fetch(`${BASE_URL}/toggle_auto_run`, { method: "POST" });
    const data = await res.json();
    console.log("Auto-Run:", data); // { auto_run: true }
}

async function toggleAutoAllow() {
    const res = await fetch(`${BASE_URL}/toggle_auto_allow`, { method: "POST" });
    const data = await res.json();
    console.log("Auto-Allow:", data); // { auto_allow: true }
}

// ── Remote Control ──────────────────────────────────

async function sendCommand(text) {
    const res = await fetch(`${BASE_URL}/send_command`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ text }),
    });
    const data = await res.json();
    console.log("Command:", data); // { status: "queued", position: 1 }
}

async function startNewChat() {
    const res = await fetch(`${BASE_URL}/start-new-chat`, { method: "POST" });
    const data = await res.json();
    console.log("New chat:", data); // { status: "queued_new_chat" }
}

async function switchChat(title) {
    const res = await fetch(`${BASE_URL}/switch_chat`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ title }),
    });
    const data = await res.json();
    console.log("Switch:", data);
}

// ── Content ─────────────────────────────────────────

async function getCommand() {
    const res = await fetch(`${BASE_URL}/get_command`);
    const data = await res.json();
    console.log("State:", data);
}

async function pushUpdate(title, content) {
    const res = await fetch(`${BASE_URL}/update`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ title, content }),
    });
    const data = await res.json();
    console.log("Update:", data); // { status: "received" }
}

// ── Stats ───────────────────────────────────────────

async function getStats() {
    const res = await fetch(`${BASE_URL}/stats`);
    const data = await res.json();
    console.log("Stats:", data);
}

async function trackAction(action) {
    const res = await fetch(`${BASE_URL}/track_action`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ action }), // "auto_run" | "auto_allow" | "remote_command"
    });
    const data = await res.json();
    console.log("Tracked:", data);
}

// ── WebSocket Stream ────────────────────────────────

function watchStream() {
    // Browser: use native WebSocket. Node: npm install ws
    const WebSocket = require("ws");
    const ws = new WebSocket("ws://localhost:9812");

    ws.on("open", () => console.log("Connected to Antigravity stream"));
    ws.on("message", (msg) => {
        const data = JSON.parse(msg);
        console.log(`[${data.title}] ${data.content.substring(0, 100)}...`);
    });
    ws.on("error", (err) => console.error("WS error:", err.message));
}

// ── Run examples ────────────────────────────────────

(async () => {
    await toggleAutoRun();
    await sendCommand("Refactor the auth module and add unit tests");
    await getStats();
    // watchStream(); // Uncomment to listen to live stream
})();

File: code_samples/python/antigravity_api.py
"""
Antigravity Automation API — Python Code Samples
Requires: pip install requests websockets
"""
import requests
import asyncio
import websockets
import json

BASE_URL = "http://localhost:5000"


# ── Automation ────────────────────────────────────────

def toggle_auto_run():
    """Toggle the Auto-Run click automation."""
    r = requests.post(f"{BASE_URL}/toggle_auto_run")
    print(r.json())  # {"auto_run": true}

def toggle_auto_allow():
    """Toggle the Auto-Allow permission automation."""
    r = requests.post(f"{BASE_URL}/toggle_auto_allow")
    print(r.json())  # {"auto_allow": true}


# ── Remote Control ────────────────────────────────────

def send_command(text: str):
    """Send a prompt/command to the active AI chat."""
    r = requests.post(f"{BASE_URL}/send_command", json={"text": text})
    print(r.json())  # {"status": "queued", "position": 1, "usage": {...}}

def start_new_chat():
    """Start a brand-new chat session."""
    r = requests.post(f"{BASE_URL}/start-new-chat")
    print(r.json())  # {"status": "queued_new_chat"}

def switch_chat(title: str):
    """Switch to an existing chat by title."""
    r = requests.post(f"{BASE_URL}/switch_chat", json={"title": title})
    print(r.json())  # {"status": "queued", "title": "...", "usage": {...}}


# ── Content ───────────────────────────────────────────

def get_command():
    """Poll for the next queued command and current state."""
    r = requests.get(f"{BASE_URL}/get_command")
    print(r.json())

def push_update(title: str, content: str):
    """Push chat content (used internally by the bridge)."""
    r = requests.post(f"{BASE_URL}/update", json={"title": title, "content": content})
    print(r.json())  # {"status": "received"}


# ── Stats ─────────────────────────────────────────────

def get_stats():
    """Retrieve cumulative usage statistics."""
    r = requests.get(f"{BASE_URL}/stats")
    print(r.json())

def track_action(action: str):
    """Track a usage action: auto_run | auto_allow | remote_command"""
    r = requests.post(f"{BASE_URL}/track_action", json={"action": action})
    print(r.json())  # {"status": "tracked", "stats": {...}}


# ── WebSocket Stream ──────────────────────────────────

async def watch_stream():
    """Stream live chat outputs via WebSocket."""
    async with websockets.connect("ws://localhost:9812") as ws:
        print("Connected to Antigravity stream...")
        while True:
            data = json.loads(await ws.recv())
            print(f"[{data['title']}] {data['content'][:100]}...")


# ── Run examples ──────────────────────────────────────

if __name__ == "__main__":
    toggle_auto_run()
    send_command("Refactor the auth module and add unit tests")
    get_stats()
    # asyncio.run(watch_stream())  # Uncomment to listen to live stream
